SuperBuild/External_CCPi-Framework.cmake

```diff
@@ -0,0 +1,129 @@
+#========================================================================
+# Author: Benjamin A Thomas
+# Author: Kris Thielemans
+# Author: Edoardo Pasca
+# Copyright 2017 University College London
+# Copyright 2017 STFC
+#
+# This file is part of the CCP PETMR Synergistic Image Reconstruction Framework (SIRF) SuperBuild.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#=========================================================================
+
+#This needs to be unique globally
+set(proj CCPi-Framework)
+
+# Set dependency list
+set(${proj}_DEPENDENCIES "")
+
+# Include dependent projects if any
+ExternalProject_Include_Dependencies(${proj} DEPENDS_VAR ${proj}_DEPENDENCIES)
+
+# Set external name (same as internal for now)
+set(externalProjName ${proj})
+
+set(${proj}_SOURCE_DIR "${SOURCE_ROOT_DIR}/${proj}" )
+set(${proj}_BINARY_DIR "${SUPERBUILD_WORK_DIR}/builds/${proj}/build" )
+set(${proj}_DOWNLOAD_DIR "${SUPERBUILD_WORK_DIR}/downloads/${proj}" )
+set(${proj}_STAMP_DIR "${SUPERBUILD_WORK_DIR}/builds/${proj}/stamp" )
+set(${proj}_TMP_DIR "${SUPERBUILD_WORK_DIR}/builds/${proj}/tmp" )
+
+if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalProjName}}" ) )
+  message(STATUS "${__indent}Adding project ${proj}")
+
+  ### --- Project specific additions here
+  set(libcilreg_Install_Dir ${SUPERBUILD_INSTALL_DIR})
+
+  #message(STATUS "HDF5_ROOT in External_SIRF: " ${HDF5_ROOT})
+  set(CMAKE_LIBRARY_PATH ${CMAKE_LIBRARY_PATH} ${SUPERBUILD_INSTALL_DIR})
+  set(CMAKE_INCLUDE_PATH ${CMAKE_INCLUDE_PATH} ${SUPERBUILD_INSTALL_DIR})
+
+  set(ENV{CIL_VERSION} 0.10.1)
```

Comment:
not sure why an env variable. and shouldn't it be in version_config.cmake ? also some hard-wired 0.10.1 below

Reply:
This is for a silly reason on the CCPi package. It should be removed from it, but it's needed for the time being.